src/components/Recipes/RecipeItem.tsx

```diff
@@ -1,14 +1,16 @@
 import "./Recipes.css"
 import './RecipeItem.css'
 import Card from '../UI/Card';
+import { useNavigate } from "react-router-dom";
 
-const RecipeItem: React.FC<{text: string, image: string}> = (props) => {
+const RecipeItem: React.FC<{recipeID: number, text: string, image: string}> = (props) => {
+  const navigate = useNavigate();
   return (
     <li>
       <Card class='recipeCard'>
         <img className="card-header" src={props.image} />
         <h2>{props.text}</h2>
-        <button className="button">
+        <button className="button" onClick={() => navigate(`/RecipePage/${props.recipeID.toString()}`)}>
```

Comment:
we need to separate this click out into a separate function -- not inline

Reply:
the reason we do this, is so we save the function in a place of memory where react doesnt have to create it every time.

passing in an anonymous function each time -- will make react rerender the component and its children each time because its getting a 'new' argument each time (new since its not saved in memory in any place)